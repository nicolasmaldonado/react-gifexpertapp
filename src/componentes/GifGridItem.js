

export const GifGridItem = ( gif ) => {

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={ gif.url } alt={ gif.title }/>
      <p>{ gif.title }</p>
    </div>
  )
}