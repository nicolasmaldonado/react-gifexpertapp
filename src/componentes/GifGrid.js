import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

  const { data: gifs, loading } = useFetchGifs( category );

  return (
    <>
      <h3>{ category }</h3>

      { loading && <p className='animate__flash'>Loading</p> }

      <div className='card-grid'>

        {
          gifs.map( (gif) => {
            return ( 
              <GifGridItem 
                key={ gif.id }
                { ...gif }
              />
            )
          })
        }

      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}