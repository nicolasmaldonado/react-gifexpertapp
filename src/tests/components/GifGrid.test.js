import { shallow } from 'enzyme';
import "@testing-library/jest-dom";
import { GifGrid } from '../../componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid />', () => {

  let wrapper;
  let category = "Hunter x Hunter"

  // beforeEach( () => {
  //   const value = "Categoria"
  //   wrapper = shallow(<GifGrid category={ value }/>);
  // });

  test('Deberia hacer match con el snapshot', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    wrapper = shallow(<GifGrid category={ category } />);
    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe mostras items cuando se cargan imagenes userFetchGifs', () => {
    
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier.cosa.jpg',
      title: 'tituloo',
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow( <GifGrid category={ category } />);

    //expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


  });
  




  
})
