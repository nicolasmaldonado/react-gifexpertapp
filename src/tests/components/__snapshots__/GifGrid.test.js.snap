// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid /> Deberia hacer match con el snapshot 1`] = `
<Fragment>
  <h3>
    Hunter x Hunter
  </h3>
  <p
    className="animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
